Race condition between Request timeout and completion

**Describe the bug**
It seems that there exist a possible race condition between completing an request and the time out procedure.

```

2020-01-06 12:56:42.303 CET
io.grpc.StatusRuntimeException: INTERNAL: Unexpected error occurred during the request processing at io.grpc.Status.asRuntimeException(Status.java:524) ~[grpc-api-1.26.0.jar:1.26.0] at io.zeebe.gateway.EndpointManager.convertThrowable(EndpointManager.java:360) ~[zeebe-gateway-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.gateway.EndpointManager.lambda$sendRequest$3(EndpointManager.java:289) ~[zeebe-gateway-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequest$3(BrokerRequestManager.java:147) ~[zeebe-gateway-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.gateway.impl.broker.BrokerRequestManager.lambda$sendRequestInternal$5(BrokerRequestManager.java:183) ~[zeebe-gateway-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.future.FutureContinuationRunnable.run(FutureContinuationRunnable.java:33) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:76) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:115) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.run(ActorThread.java:195) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] Caused by: java.util.concurrent.TimeoutException: Request timed out after PT1M at io.zeebe.transport.impl.AtomixClientTransportAdapter.timeoutFuture(AtomixClientTransportAdapter.java:120) ~[zeebe-transport-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.transport.impl.AtomixClientTransportAdapter.lambda$sendRequestWithRetry$0(AtomixClientTransportAdapter.java:57) ~[zeebe-transport-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] ... 6 more
2020-01-06 12:56:44.812 CET
java.lang.IllegalStateException: Cannot complete future, the future is already completed exceptionally with 'java.util.concurrent.TimeoutException: Request type command-api-2 timed out in 61254 milliseconds' at io.zeebe.util.sched.future.CompletableActorFuture.completeExceptionally(CompletableActorFuture.java:184) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.future.CompletableActorFuture.completeExceptionally(CompletableActorFuture.java:191) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.transport.impl.AtomixClientTransportAdapter.handleResponse(AtomixClientTransportAdapter.java:107) ~[zeebe-transport-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.transport.impl.AtomixClientTransportAdapter.lambda$tryToSend$3(AtomixClientTransportAdapter.java:84) ~[zeebe-transport-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorJob.invoke(ActorJob.java:73) ~[zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorJob.execute(ActorJob.java:39) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorTask.execute(ActorTask.java:115) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.executeCurrentTask(ActorThread.java:107) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.doWork(ActorThread.java:91) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT] at io.zeebe.util.sched.ActorThread.run(ActorThread.java:195) [zeebe-util-0.22.0-SNAPSHOT.jar:0.22.0-SNAPSHOT]
```

**To Reproduce**

Just run a benchmark in our test cluster.

**Expected behavior**
No exception.


**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

--- a/transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java
+++ b/transport/src/main/java/io/zeebe/transport/impl/AtomixClientTransportAdapter.java
@@ -86,6 +86,10 @@ public final class AtomixClientTransportAdapter extends Actor implements ClientT
 
   private void handleResponse(
       final RequestContext requestContext, final byte[] response, final Throwable errorOnRequest) {
+    if (requestContext.isDone()) {
+      return;
+    }
+
     final var currentFuture = requestContext.getCurrentFuture();
     if (errorOnRequest == null) {
       final var responseBuffer = new UnsafeBuffer(response);

